test/c2llvm2mlir/simple_loop/test.mlir

```diff
@@ -15,3 +15,55 @@
 // Verifies the output values are the same for the original and re-compiled kernel.
 // CHECK: output: [[OUTPUT:[0-9]+\.[0-9]+]]
 // CHECK: output: [[OUTPUT]]
+
+// Tests LLVM to NEURA lowering.
+// RUN: clang++ -S -emit-llvm -O3 -fno-unroll-loops -fno-vectorize -ffp-contract=off kernel.cpp -o %t-kernel.ll
+// RUN: mlir-translate --import-llvm %t-kernel.ll -o %t-kernel.mlir
+
+// RUN: mlir-neura-opt --assign-accelerator \
+// RUN:   --lower-llvm-to-neura \
+// RUN:   --promote-func-arg-to-const \
+// RUN:   --fold-constant \
+// RUN:   --canonicalize-live-in \
+// RUN:   --leverage-predicated-value \
+// RUN:   --transform-ctrl-to-data-flow \
+// RUN:   --view-op-graph \
+// RUN:   --architecture-spec=../../arch_spec/architecture.yaml \
+// RUN:   --insert-data-mov %t-kernel.mlir -o %t-kernel-neura.mlir
+// RUN: FileCheck %s --check-prefix=CHECK-LLVM2NEURA < %t-kernel-neura.mlir
+
+// RUN: mlir-neura-opt --assign-accelerator \
+// RUN:   --lower-llvm-to-neura \
+// RUN:   --promote-func-arg-to-const \
+// RUN:   --fold-constant \
+// RUN:   --canonicalize-live-in \
+// RUN:   --leverage-predicated-value \
+// RUN:   --transform-ctrl-to-data-flow \
+// RUN:   --view-op-graph \
+// RUN:   --architecture-spec=../../arch_spec/architecture.yaml \
+// RUN:   --insert-data-mov \
+// RUN:   --map-to-accelerator="mapping-strategy=heuristic backtrack-config=customized=5,3" %t-kernel.mlir -o %t-kernel-mapped.mlir
+// RUN: FileCheck %s --check-prefix=CHECK-LLVM2NEURA-MAP < %t-kernel-mapped.mlir
+
+// CHECK-LLVM2NEURA: accelerator = "neura"
+// CHECK-LLVM2NEURA: dataflow_mode = "predicate"
+// CHECK-LLVM2NEURA: neura.phi
+// CHECK-LLVM2NEURA: neura.gep
```

Comment:
Can we also check whether there exists `operand_segment_sizes` in this op? which is generated because you use `AttrSizedOperandSegments`.